src/editors/containers/VideoEditor/components/VideoSettingsModal/components/HandoutWidget/index.jsx

```diff
@@ -0,0 +1,125 @@
+  const fileInput = hooks.fileInput({ fileSizeError });
+  const handoutName = hooks.parseHandoutName({ handout });
+  const downloadLink = getHandoutDownloadUrl({ handout });
+
+  return (
+    <CollapsibleFormWidget
+      isError={Object.keys(error).length !== 0}
+      title={intl.formatMessage(messages.titleLabel)}
+      subtitle={handoutName}
+    >
+      <ErrorAlert
+        dismissError={fileSizeError.dismiss}
+        hideHeading
+        isError={fileSizeError.show}
+      >
+        <FormattedMessage {...messages.fileSizeError} />
+      </ErrorAlert>
+      <UploadErrorAlert message={messages.uploadHandoutError} />
+      <FileInput fileInput={fileInput} />
+      {handout ? (
+        <Card>
+          <Card.Header
+            className="mt-1"
+            subtitle={handoutName}
+            actions={(
+              <Dropdown>
+                <Dropdown.Toggle
+                  id="dropdown-toggle-with-iconbutton-video-transcript-widget"
+                  as={IconButton}
+                  src={MoreHoriz}
+                  iconAs={Icon}
+                  variant="primary"
+                  alt="Actions dropdown"
+                />
+                <Dropdown.Menu className="video_handout Action Menu">
+                  <Dropdown.Item
+                    key="handout-actions-replace"
+                    onClick={fileInput.click}
+                  >
+                    <FormattedMessage {...messages.replaceHandout} />
+                  </Dropdown.Item>
+                  <Dropdown.Item key="handout-actions-download" target="_blank" href={downloadLink}>
+                    <FormattedMessage {...messages.downloadHandout} />
+                  </Dropdown.Item>
+                  <Dropdown.Item key="handout-actions-delete" onClick={() => updateField({ handout: null })}>
+                    <FormattedMessage {...messages.deleteHandout} />
+                  </Dropdown.Item>
+                </Dropdown.Menu>
+              </Dropdown>
+            )}
+          />
+        </Card>
+      ) : (
+        <Stack gap={3}>
+          <FormattedMessage {...messages.addHandoutMessage} />
+          <Button iconBefore={FileUpload} onClick={fileInput.click} variant="link">
+            <FormattedMessage {...messages.uploadButtonLabel} />
+          </Button>
+        </Stack>
+      )}
+    </CollapsibleFormWidget>
+  );
+};
+
+HandoutWidget.propTypes = {
+  // injected
+  intl: intlShape.isRequired,
+  // redux
+  handout: PropTypes.shape({}).isRequired,
+  updateField: PropTypes.func.isRequired,
+  isUploadError: PropTypes.bool.isRequired,
+  getHandoutDownloadUrl: PropTypes.func.isRequired,
+};
+export const mapStateToProps = (state) => ({
+  handout: selectors.video.handout(state),
+  getHandoutDownloadUrl: selectors.video.getHandoutDownloadUrl(state),
+});
+
+export const mapDispatchToProps = (dispatch) => ({
+  updateField: (stateUpdate) => dispatch(actions.video.updateField(stateUpdate)),
```

Comment:
nit: stateUpdate implies that this is updating the whole state `payload` is the naming convention we have adopted for modifications to a state (like in a reducer)